### PR TITLE
major: Move processors to separate module

### PR DESCRIPTION
**Pull request by timyates**
Relates to #1588

### CODE REVIEW

**Comment by timyates** [comment]
The coverage is because these are annotation processors, so we have tests that they work, but I don't believe they work during the phase that sonar is watching for coverage

File: security-oauth2/build.gradle.kts [hunk 1]
```diff
@@ -17,8 +18,10 @@ dependencies {
     compileOnly(mn.jackson.databind)
     compileOnly(mnSession.micronaut.session)
     implementation(mnReactor.micronaut.reactor)
+
     testImplementation(mnSerde.micronaut.serde.jackson)
     testImplementation(projects.micronautSecuritySession)
+    testImplementation(projects.micronautSecurityProcessor)
```
Comment (sdelamo) [inline security-oauth2/build.gradle.kts:24]: should this be `testCompileOnly` for Groovy?

File: security/build.gradle.kts [hunk 2]
```diff
@@ -20,16 +20,19 @@ dependencies {
     compileOnly(mn.micronaut.http.server)
     compileOnly(mn.micronaut.management)
     compileOnly(mn.jackson.databind)
+
+    testAnnotationProcessor(projects.micronautSecurityProcessor)
+    testAnnotationProcessor(mn.micronaut.inject.java)
+
     testCompileOnly(mnData.micronaut.data.processor)
+    testImplementation(projects.micronautSecurityProcessor)
```
Comment (sdelamo) [inline security/build.gradle.kts:28]:
```suggestion
    testCompileOnly(projects.micronautSecurityProcessor)
```

File: security-oauth2/build.gradle.kts (hunk 1 above)
Comment (sdelamo) [inline security-oauth2/build.gradle.kts:24]:
```suggestion
    testCompileOnly(projects.micronautSecurityProcessor)
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**